pkg/daemon/ceph/osd/volume.go

```diff
@@ -960,55 +960,57 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 	//
 	// isCephEncryptedBlock() returns false if the disk is not a LUKS device with:
 	// Device /dev/sdc is not a valid LUKS device.
-	if isCephEncryptedBlock(context, cephfsid, block) {
-		childDevice, err := sys.ListDevicesChild(context.Executor, block)
-		if err != nil {
-			return nil, err
-		}
+	if block != "" {
```

Comment:
Why would the block not be set?

Reply:
It depends on how `GetCephVolumeRawOSDs()` is called. Sometimes it is called against a device, sometimes it's not. For instance, in the cleanup scenario, we don't pass any `block` because we are looking for **all** the OSDs present on the machine. On the other hand, the PVC scenario always uses the PVC block as a `block` to check whether the disk is an OSD or not.

Reply:
Thanks for the explanation, want to add a comment about that?